### DISCUSSION TITLE
Trouble Running a Javascript File

### DISCUSSION THREAD

**Discussion by mrghubba** [ConsenSys-archive/truffle#5447] [General Q&A]
Hi there.
I'm trying to utilize the 0x API for swapping tokens. 
I tried running this Javascript file:
**https://github.com/0xProject/0x-api-starter-guide-code/blob/master/src/swap-contract.js**

But I got this error:
https://paste.pics/I2YBG

Does anyone here know why?

**Comment by kevinbluer** [comment]
Hey @mrghubba! I just had a quick walk through some of the steps in the project's [README](https://github.com/0xProject/0x-api-starter-guide-code#installation) and was largely able to get it working. Just to confirm, did you follow the installation steps (specifically getting the dependencies installed)? Also did you run `swap-contract.js` via the associated scripts in the `package.json`, e.g. https://github.com/0xProject/0x-api-starter-guide-code#contract-swap?

**Reply by mrghubba** [reply]
Hi. I wasn't able to complete the first step of README. When I tried to clone the repository, I got this error: https://paste.pics/I7LLI

**Comment by mrghubba** [comment]
I'll be honest. I did not follow the steps from README as shown here:
https://paste.pics/I70RC

I was reluctant to do so given the experience I had before with my ARM64
system trying to get the right version of NodeJS (if you recall).
To avoid all that, I wanted to see if I could run everything under the same
IDE to simplify things.  I figured that if I'm still using Ganache, using a
fork and an IDE that claims to run scripts, it would still work. But from
what you're saying, I really do have to follow the README instructions.

[image: width=]
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>
